### ISSUE TITLE
Home directory option for private.xml

### ISSUE THREAD

**Issue by pzuraq** [tekezo/Karabiner#377]
I would like an option to include a `private.xml` config file from the home directory, something like:

```
~/.karabiner/private.xml
```

This would make it a lot easier to include `private.xml` configs in dot repositories and generally find, modify, and share the config. I'd be up for adding the functionality and submitting a PR if it seems like something the project would like to add.

**Comment by tekezo** [comment]
How about <include>?
https://pqrs.org/osx/karabiner/xml.html.en#include

**Comment by pzuraq** [comment]
That would work, it would be nice if that functionality came out of the box though. Consider a fresh install, would be very nice to install karabiner and git clone my settings and be done.

Perhaps private.xml could be modified to import a file by default? That would be very simple to add.

**Comment by tekezo** [comment]
Hmm, how do you import settings?
Either copying org.pqrs.Karabiner.plist or importing exported settings require you to an extra action.
I think you can overwrite ~/Library/Application\ Support/Karabiner/private.xml at the same time.

**Comment by pzuraq** [comment]
I import settings by cloning my settings repository directly into my home folder, makes everything much easier for setup. I could create a script to move the file there, sort of a pain but at least it's still tracked and way easier to modify with the `include` statement.

**Comment by tekezo** [comment]
Unlike UNIX apps, OS X apps store their configuration into ~/Library/Preferences/.
Karabiner follows the OS X way.
So, you should restore configuration when initial cloning.
(Or import an exported configuration: https://pqrs.org/osx/karabiner/document.html.en#export )

Therefore, if we want to restore settings by cloning dot files,
we also need to change Karabiner to follow the UNIX configuration way.
(eg. save settings into ~/.karabiner/rc )

The change needs a time. So, please run extra script when initial cloning at the moment.

**Comment by highvoltag3** [comment]
I agree with @tekezo tho one thing that would be nice (and why I got here, I was about to open a ticket for it), it'd be nice if Karabiner supported reading from an alias (symlink) say I want to keep my private.xml in dropbox so that I can use the same file both in my main iMac and my laptop. 

Alfred for example and some other apps have "native support" well they basically allow for what's requested here, but in cases where that's not actually possible a work around is a symlink to the folder in your dropbox, iCloud, etc.. I find that would be a really interesting feature for Karabiner.

I just tried this but it seems like the app refuses to read the .xml symlink

<img width="664" alt="pasted_image_11_1_15__12_33_am" src="https://cloud.githubusercontent.com/assets/694585/10867744/5f2dda60-8030-11e5-8ef5-59d2d61be0f5.png">

**Comment by pzuraq** [comment]
I was actually doing just this to get my private.xml to link to my dotrepo, but in the latest update it stopped working.



— Chris G.

On Sat, Oct 31, 2015 at 11:35 PM, Dario J Novoa V.

**Comment by tekezo** [comment]
If you do not use `<include>` yet, please use it.
https://pqrs.org/osx/karabiner/xml.html.en#dropbox

**Comment by highvoltag3** [comment]
@tekezo is 100% right thanks BTW.

Using `<include>` works perfectly with the file in dropbox, and to the end purpose it's the same as having the symlink file.

**Comment by pzuraq** [comment]
Personally don't like using <include> because it's not as easy to automate. My dotrepo it setup using dotbot, it works by symlinking all of the config files to various locations. Copying a standard config file is a bit more of a complicated task, several steps and then you have to diff it to make sure that no changes were made. 



— Chris G.

**Comment by undeadlegion** [comment]
@pzuraq I'm unfamiliar with dotbot, but I currently store my config file in dropbox and replace private.xml with a sym link
